extensions/schema-compare/src/utils.ts

```diff
@@ -40,12 +41,46 @@ export function getTelemetryErrorType(msg: string): string {
  */
 export function getEndpointName(endpoint: azdata.SchemaCompareEndpointInfo): string {
 	if (!endpoint) {
-		return undefined;
+		return ' ';
 	}
 
 	if (endpoint.endpointType === azdata.SchemaCompareEndpointType.Database) {
+		if (!endpoint.serverName && endpoint.connectionDetails) {
+			endpoint.serverName = endpoint.connectionDetails['serverName'];
+		}
 		return `${endpoint.serverName}.${endpoint.databaseName}`;
 	} else {
 		return endpoint.packageFilePath;
 	}
+}
+
+function connectionInfoToConnectionProfile(details: azdata.ConnectionInfo): azdata.IConnectionProfile {
+	return {
+		serverName: details['serverName'],
+		databaseName: details['databaseName'],
+		authenticationType: details['authenticationType'],
+		providerName: 'MSSQL',
+		connectionName: '',
+		userName: details['userName'],
+		password: details['password'],
+		savePassword: false,
+		groupFullName: undefined,
+		saveProfile: true,
+		id: undefined,
+		groupId: undefined,
+		options: details['options']
+	};
+}
+
+export async function verifyConnectionAndGetOwnerUri(endpoint: azdata.SchemaCompareEndpointInfo): Promise<string> {
```

Comment:
+1 on this :)

I was thinking that now there are enough utils functions that we should move those to a separate file... this is getting too long for readability (not in this PR)